### PR TITLE
Upgrade airbase version to 46.

### PR DESCRIPTION
**Pull request by kokosing**
Upgrade airbase version to 46.

 - As new airbase comes with new version of testng some tests had to be
refactored as they started to fail.
 - Upgrade also airlift, discover-server and resolver to versions which
 also are using airbase at version 46, as with older version there were
 a dependency conflict.

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/util/Numbers.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        return number instanceof Double || number instanceof Float;
+    }
+
+    public static boolean isNan(Object number)
+    {
+        return number.equals(Double.NaN) || number.equals(Float.NaN);
+    }
+
+    private Numbers()
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/util/Numbers.java]: Put the constructor at the top of the class and format like this:
```
private Numbers() {}

File: presto-main/src/test/java/com/facebook/presto/util/Numbers.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+public class Numbers
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/util/Numbers.java]: Utility classes should be final

**Comment by electrum** [comment]
Commit titles should not end with a period (they are not always sentences)

**Comment by kokosing** [comment]
Rebased, comments were addressed. I removed SNAPSHOT version of airbase. To make this patch be eligible for merge:
 - https://github.com/airlift/airlift/pull/360 needs to be merged and then airlift  released
 - https://github.com/airlift/resolver/pull/10 needs to be merged and then resolver released
 - https://github.com/airlift/discovery/pull/38 needs to be merged and then discovery  released

**Comment by kokosing** [comment]
Rebased, SNAPSHOT versions of dependencies removed.

**Comment by martint** [comment]
Thanks, I'll merge it after we're done cutting the current release.

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**